selfdrive/manager/manager.py

```diff
@@ -99,13 +100,13 @@ def manager_prepare(spinner=None):
   os.chdir(os.path.dirname(os.path.abspath(__file__)))
 
   total = 100.0 - (0 if PREBUILT else MAX_BUILD_PROGRESS)
-
   for i, p in enumerate(managed_processes.values()):
-    perc = (100.0 - total) + total * (i + 1) / len(managed_processes)
-
+    p.prepare()
     if spinner:
+      perc = (100.0 - total) + total * (i + 1) / len(managed_processes)
+      if perc == 100:
+        time.sleep(1 / 50.)
```

Comment:
Let me test this, I actually think it won't work since it will just be at the last progress+1

Reply:
the sleep should be reverted

Reply:
Done. If it's okay, sending 100% once more right before launching the processes makes the bar finish and show for about the same amount of time I had it sleep for before.

Reply:
I don't want to merge in any hacks like this. feel free to fix it properly in the old spinner and open a PR, but we'll spend some time making this nice when we ship the Qt UI

Reply:
Sounds good. I tried to fix it in the spinner C code but couldn't get anywhere. I'll open a PR if I do though